tasks/jsinspect.js

```diff
@@ -45,7 +46,16 @@ module.exports = function(grunt) {
       suppress: options.suppress
     });
 
-    if (options.failOnMatch) {
+    if (typeof options.failOnMatch === 'number') {
+      // Handle failOnMatch threshold.
+      inspector.on('match', function() {
+        nbOfMatches++;
+        if (nbOfMatches >= options.failOnMatch) {
+          taskSucceeded = false;
+        }
+      });
+    } else if (options.failOnMatch) {
```

Comment:
How about making this `options.failOnMatch === true`, in order to force it as boolean?